symfony/plugins/orangehrmAdminPlugin/Api/WorkShiftAPI.php

```diff
@@ -0,0 +1,257 @@
+        return new ParamRuleCollection(
+            ...$this->getSortingAndPaginationParamsRules(WorkShiftSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $workShift = $this->getWorkShiftService()->getWorkShiftById($id);
+        $this->throwRecordNotFoundExceptionIfNotExist($workShift, WorkShift::class);
+        return new EndpointResourceResult(WorkShiftDetailedModel::class, $workShift);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @throws DaoException
+     */
+    public function create(): EndpointResult
+    {
+        $workShift = $this->saveWorkShift();
+        return new EndpointResourceResult(WorkShiftModel::class, $workShift);
+    }
+
+    /**
+     * @return WorkShift
+     * @throws RecordNotFoundException
+     * @throws DaoException
+     */
+    public function saveWorkShift(): WorkShift
+    {
+        $workShiftId = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_ID
+        );
+
+        $workShiftName = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        $workShiftHoursPerDay = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_HOURS_PER_DAY
+        );
+        $workShiftStartTime = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_START_TIME
+        );
+        $workShiftEndTime = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_END_TIME
+        );
+        $empNumbers = $this->getRequestParams()->getArray(
+            RequestParams::PARAM_TYPE_BODY,
+            CommonParams::PARAMETER_EMP_NUMBERS
+        );
+
+        if (!empty($workShiftId)) {
+            $workShift = $this->getWorkShiftService()->getWorkShiftById($workShiftId);
+            if ($workShift == null) {
+                throw new RecordNotFoundException();
+            } else {
+                $workShift->setName($workShiftName);
+                $workShift->setHoursPerDay($workShiftHoursPerDay);
+                $workShift->setStartTime(new DateTime($workShiftStartTime));
+                $workShift->setEndTime(new DateTime($workShiftEndTime));
+                return $this->getWorkShiftService()
+                    ->getWorkShiftDao()
+                    ->updateWorkShift($workShift, $empNumbers);
+            }
```

Comment:
remove this else. since all property setters call below as well

Reply:
done